src/scripts/main.js

```diff
@@ -1,3 +1,46 @@
 'use strict';
 
 // write code here
+const listEmployee = document.querySelectorAll('li');
+
+function parseSalary(salaryString) {
+  const cleanedString = salaryString.replace(/[^0-9.-]+/g, '');
+
+  // Перевіряємо, чи це дійсно валідне число
+  if (!isNaN(cleanedString) && cleanedString.trim() !== '') {
+    return parseFloat(cleanedString);
+  } else {
+    return null;
+  }
+}
+
+function sortList(list) {
+  return list.sort((a, b) => {
+    // Отримуємо зарплату з data-атрибутів і перетворюємо в число
+    const salaryA = parseSalary(a.dataset.salary);
+    const salaryB = parseSalary(b.dataset.salary);
+
+    // Сортуємо за спаданням зарплати
+    return salaryB - salaryA;
+  });
```

Comment:
The `sortList` function is currently modifying the original list passed in as a parameter because `Array.prototype.sort()` sorts the array in place. This can lead to unexpected behavior if the original list is used elsewhere in the code. To prevent this, you should create a new array that is a copy of the original before sorting it. You can do this using the spread operator [...list].